docs/SIL.rst

```diff
@@ -17,8 +17,10 @@ the Swift programming language. SIL accommodates the following use cases:
   such as definitive initialization of variables and constructors, code
   reachability, switch coverage.
 - High-level optimization passes, including retain/release optimization,
-  dynamic method devirtualization, closure inlining, memory allocation promotion,
-  and generic function instantiation.
+  dynamic method devirtualization, closure inlining, promoting heap allocations
+  to stack allocations, promoting stack allocations to SSA registers, scalar
```

Comment:
`promoting stack allocations to SSA registers` => `promoting memory allocations to SSA registers`? Your thoughts? Once we resolve that LGTM.

Reply:
I wrote it so that is more consistent with these lines: https://github.com/apple/swift/blame/2255e81ad0e12472c31fbf3b01231321a35881e6/docs/SIL.rst#L83-L85

Thoughts?

Reply:
Ok. I am fine with that. I just got nervous that someone could extrapolate that we didn’t handle the heap case. I think I was being to fearful